### ISSUE TITLE
echo cancel suport 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [BinaryBlue/imsdroid#250]
```
a) Before posting your issue you MUST answer to the questions otherwise it
will be rejected (invalid status) by us
b) Please check the issue tacker to avoid duplication
c) Please provide network capture (wireshark) or Android log (DDMS output)
if you want quick response

What steps will reproduce the problem?
1.
2.
3.

What is the expected output? What do you see instead?

Hi, bossiel, i have seen that imsdroid has add echo cancel model, but it seem 
that is has no effect at all,
do you hava any hint or what do i need set ?

thanks . 

What version of the product are you using? On what operating system?


Please provide any additional information below.


```

Original issue reported on code.google.com by `openser@yeah.net` on 27 Jun 2011 at 9:28

**Comment by GoogleCodeExporter** [comment]
```
Yes, echo cancellation is now support in Doubango framework but disable on 
Android because of lack of support for low latency audio.
To enable echo cancellation, change these lines: 
http://code.google.com/p/imsdroid/source/browse/branches/2.0/android-ngn-stack/s
rc/org/doubango/ngn/media/NgnProxyPluginMgr.java#56

to:
MediaSessionMgr.defaultsSetEchoSuppEnabled(true);
you will also need large echo tail (because of audio delay on Android). To set 
the echo tail:
MediaSessionMgr.defaultsSetEchoTail(echo_tail); // echo_tail: number of packets
```

Original comment by `boss...@yahoo.fr` on 27 Jun 2011 at 12:22
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Hi boss...,
Please tell mw how to set right echo_tail value? THanks so much.
```

Original comment by `hoangdinhtrung` on 28 Jun 2011 at 8:43

**Comment by GoogleCodeExporter** [comment]
```
as android has latency in audio, and speex fuction 
SpeexEchoState *speex_echo_state_init(int frame_size, int filter_length);
the desc imply that the filter_length shuld be 100~500ms.
```

Original comment by `openser@yeah.net` on 28 Jun 2011 at 10:27

**Comment by GoogleCodeExporter** [comment]
```
thanks for your time bossiel,
i have seach a lot about echo in android, and some othe sip softphone has many 
discuss adout this , and may this is a bug in andoid , you may check link 
below, they 
discuss something about Audio mode for SIP calls in android ,
in file setSpeakerphoneOn , i see that some place relate to this, it seems that 
we should use  MODE_NORMAL instead if MODE_IN_CALL,
 may this help you, thanks.

http://code.google.com/p/csipsimple/issues/detail?id=119#c38
```

Original comment by `openser@yeah.net` on 29 Jun 2011 at 12:47

**Comment by GoogleCodeExporter** [comment]
```
sorry, it is setSpeakerphoneOn function that in file  ngnproxaudioconsumer.java 
：）
```

Original comment by `openser@yeah.net` on 29 Jun 2011 at 12:49

**Comment by GoogleCodeExporter** [comment]
```
we have resole it. use speex, thanks.
```

Original comment by `openser@yeah.net` on 12 Jul 2011 at 5:36

**Comment by GoogleCodeExporter** [comment]
```
Fixed by IMSDroid 2.0.431
```

Original comment by `boss...@yahoo.fr` on 13 Jul 2011 at 11:57
- Changed state: **Fixed**
